test/stepDefinitions/definitionFileSteps.js

```diff
@@ -30,9 +30,23 @@ defineSupportCode(function ({ Given, When, Then, Before, After }) {
     Data.optionalFields = [{fieldType: 'text', fieldId: 'CollectionComplexField_0_AddressLine1'},
                             {fieldType: 'text', fieldId: 'CollectionComplexField_0_AddressLine2'},
                             {fieldType: 'text', fieldId: 'CollectionComplexField_0_AddressLine3'},
-                            {fieldType: 'text', fieldId: 'CollectionComplexField_0_Country'}];
+                            {fieldType: 'text', fieldId: 'CollectionComplexField_0_Country'},
+                            {fieldType: 'text', fieldId: 'CollectionComplexField_0_SecondAddress_SecondLine1'}];
     });
 
+  Given(/^a case type containing conditional data of collection of complex types exists$/, function() {
+    Data.jurisdiction = 'Auto Test 1';
+    Data.caseType = 'Complex in Coll in Complex';
+    Data.event = 'Create Collection of Complex';
+    Data.optionalFields = [{fieldType: 'text', fieldId: 'CollectionComplexField_0_AddressLine1'},
+      {fieldType: 'text', fieldId: 'CollectionComplexField_0_AddressLine2'},
+      {fieldType: 'text', fieldId: 'CollectionComplexField_0_AddressLine3'},
+      {fieldType: 'US', fieldId: 'CollectionComplexField_0_Country'},
```

Comment:
This is filled out wrong, you are entering 'values' as 'FieldTypes' if you want to define value here too do it like so ` {fieldType: 'text', fieldId: 'MySchool_Class_0_ClassName', value: 'Analytical Maths'},`